### PR TITLE
Update controller tests; Prevent setting modification when a proposal is still active

### LINKED ISSUE #199: Full coverage on controller actions test
Add tests for the set/reset of each parameter in the space contract by the space controller.

### LINKED ISSUE #275: Add test for quorum
We don't have any test to check for `Quorum has not been reached` message :)

### PR DESCRIPTION
**Pull request by pscott**
This PR does two things:
- Updates the controller action tests, making sure it's working as intended, adding some sweet regression tests
- Prevents settings to be modified when a proposal is still active (temporary fix to #116 ) and adds some new tests to check it :)

Replaces #208 (I used @Orland0x 's branch as a base branch)
Closes #199 
Closes #275

### CODE REVIEW

File: test/starknet/ActiveProposal.test.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { expect } from 'chai';
+import { starknet, ethers } from 'hardhat';
+import { vanillaSetup } from '../shared/setup';
+import { StarknetContract, Account } from 'hardhat/types';
+import { utils } from '@snapshot-labs/sx';
+import { getProposeCalldata } from '@snapshot-labs/sx/dist/utils/encoding';
```
Comment (Orland0x) [inline test/starknet/ActiveProposal.test.ts]: can just access from `utils.encoding.getProposeCalldata`

**Comment by Orland0x** [comment]
Most settings are stored at the proposal level, so they can be changed even if a proposal is active right? whats the motivation to prevent that?

**Comment by pscott** [comment]
> Most settings are stored at the proposal level, so they can be changed even if a proposal is active right? whats the motivation to prevent that?

I'm not sure I understand the question. For proposal level settings, we don't need to prevent the modification, since they're already stored at the proposal level. Modifying those will only impact the following proposals, not those already active :)

**Comment by Orland0x** [comment]
> > Most settings are stored at the proposal level, so they can be changed even if a proposal is active right? whats the motivation to prevent that?
> 
> I'm not sure I understand the question. For proposal level settings, we don't need to prevent the modification, since they're already stored at the proposal level. Modifying those will only impact the following proposals, not those already active :)

eg adding an executor, we prevent that despite executors being stored at proposal level

**Review by Orland0x — APPROVED** [review]

**Pull request merged**